pkg/chartutil/requirements.go

```diff
@@ -266,3 +268,128 @@ func ProcessRequirementsEnabled(c *chart.Chart, v *chart.Config) error {
 
 	return nil
 }
+
+// pathToMap creates a nested map given a YAML path in dot notation
+func pathToMap(path string, data map[string]interface{}) map[string]interface{} {
```

Comment:
- [ ] I found this function hard to follow, so I rewrote it to understand it better.  
  I interpret as pointing the last path component to `data`, and each preceding path component to the next, where each component is represented as a map with a single entry, the key being the path component, and the value being the map holding the successor key.
  Does the following achieve the same goal?
```go
func pathToMap(path string, data map[string]interface{}) map[string]interface{} {
	components := strings.Split(path, ".")
/* What should happen if the path is empty?
	if len(components) == 0 {
		return nil
	}
*/
	sort.Sort(sort.Reverse(sort.StringSlice(components)))
	m := data
	for _, c := range components {
		m = map[string]interface{}{c: m}
	}
	return m
}
```